#! /bin/sh /usr/share/dpatch/dpatch-run
## 07-update_aclocal.m4.dpatch by Sven Mueller <debian@incase.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: update aclocal.m4 and configure.in

@DPATCH@
--- a/configure.in
+++ b/configure.in
@@ -51,6 +51,11 @@ AC_PREREQ([2.54])
 AC_CONFIG_HEADER(config.h)
 AC_CANONICAL_HOST
 
+dnl Useful for maintainer mode stuff
+WARNERROR=-W
+AC_ARG_ENABLE(warnings-are-errors,[  --enable-warnings-are-errors   add -Werror to most gcc calls], [ if test "$enableval" != "no" ; then WARNERROR=-Werror ; fi ])
+AC_SUBST(WARNERROR)
+
 dnl Useful hook for distributions
 AC_ARG_WITH(extraident,[  --with-extraident=STRING   use STRING as extra version information],
 	[AC_DEFINE_UNQUOTED(EXTRA_IDENT,"$withval", [Extra version information for imap/version.h])])
@@ -94,10 +99,12 @@ AC_AIX
 AC_ISC_POSIX
 AC_PROG_AWK
 AC_C_CONST
-AC_SYS_LONG_FILE_NAMES
-if test $ac_cv_sys_long_file_names = no; then
-	AC_MSG_ERROR(The Cyrus IMAPD requires support for long file names)
-fi
+dnl fakeroot sometimes fails this test
+dnl and Debian always supports long filenames anyway
+dnl AC_SYS_LONG_FILE_NAMES
+dnl if test $ac_cv_sys_long_file_names = no; then
+dnl	AC_MSG_ERROR(The Cyrus IMAPD requires support for long file names)
+dnl fi
 AC_C_INLINE
 
 CMU_C___ATTRIBUTE__
@@ -1202,7 +1209,11 @@ enum {
     SQUAT_ENGINE = 1,
 
     /* should we have long LMTP error messages? */
-    LMTP_LONG_ERROR_MSGS = 1
+    LMTP_LONG_ERROR_MSGS = 1,
+
+    /* default time to wait, in seconds, before giving up
+     * trying to lock something */
+    LOCK_GIVEUP_TIMER_DEFAULT = 100
 };
 
 #endif /* _CYRUS_IMAPD_CONFIG_H_ */
